.github/workflows/release-porting-guide.yml

```
@@ -0,0 +1,94 @@
+---
+name: Ansible porting guide creation
+on:
+  workflow_dispatch:
+    inputs:
+      ansible-build-data-branch:
+        description: >-
+          Release Branch name from the Ansible Build data PR.
+          (e.g. `refs/pull/1/merge`)
+        required: true
+      ansible-version:
+        description: >-
+          Exact release version. For example, 12.1.0
+        required: true
+
+jobs:
+  upload-porting-guide:
+    name: Extract the porting guide
+    runs-on: ubuntu-latest
+    env:
+      GIT_BRANCH: "release/porting-guide-${{ inputs.ansible-version }}"
+      CI_COMMIT_MESSAGE: >-
+        Add the Ansible community ${{ inputs.ansible-version }} porting guide
+
+    steps:
+      - name: Compute derived context
+        id: context
+        env:
+          ANSIBLE_VERSION_FULL: ${{ inputs.ansible-version }}
+        run: >-
+          echo ansible-version-major="${ANSIBLE_VERSION_FULL%%.*}"
+          >> "${GITHUB_OUTPUT}"
+        shell: bash --noprofile --norc -O extglob -eEuo pipefail {0}
+
+      - name: Check out this repo src
+        uses: actions/checkout@v4
+      - name: Check out ansible-build-data
+        uses: actions/checkout@v4
+        with:
+          repository: ansible-community/ansible-build-data
+          ref: ${{ inputs.ansible-build-data-branch }}
+          path: ansible-build-data
+
+      - name: Copy the RST file to the correct path
+        env:
+          ANSIBLE_VERSION_MAJOR: ${{ steps.context.outputs.ansible-version-major }}
+        run: >-
+          cp -v
+          "ansible-build-data/${ANSIBLE_VERSION_MAJOR}/porting_guide_${ANSIBLE_VERSION_MAJOR}.rst"
+          docs/docsite/rst/porting_guides/
+
+      - name: Set up git
+        run: |
+          git switch --create "${GIT_BRANCH}"
+          ACTOR_NAME="$(curl -s https://api.github.com/users/"${GITHUB_ACTOR}" | jq --raw-output '.name // .login')"
+          git config --global user.name "${ACTOR_NAME}"
+          git config --global user.email "${GITHUB_ACTOR_ID}+${GITHUB_ACTOR}@users.noreply.github.com"
+
+      - name: Add the porting guide
+        env:
+          ANSIBLE_VERSION_MAJOR: ${{ steps.context.outputs.ansible-version-major }}
+        run: git add docs/docsite/rst/porting_guides/"porting_guide_${ANSIBLE_VERSION_MAJOR}.rst"
+
+      - name: Commit the porting guide
+        run: >-
+          git diff-index --quiet HEAD ||
+          git commit -m "${CI_COMMIT_MESSAGE}"
+
+      - name: Push to the repo
+        run: git push origin "${GIT_BRANCH}"
+
+      - name: Create the porting guide PR as draft
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+          ANSIBLE_VERSION_FULL: ${{ inputs.ansible-version }}
+          PR_BODY_MESSAGE: >-
```

As Felix pointed out, we shouldn't smash everything in one line:
```suggestion
          PR_BODY_MESSAGE: |-
```

Here's where you can experiment with block scalars interactively: https://yaml-multiline.info